pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -668,54 +653,42 @@ func parseResourceGroupHashFromAnnotation(s *fleetv1beta1.ClusterResourceSnapsho
-			return nil, controller.NewUnexpectedBehaviorError(err)
-		}
-		ri := fleetv1beta1.ResourceIdentifier{
-			Group:     obj.GroupVersionKind().Group,
-			Version:   obj.GroupVersionKind().Version,
-			Kind:      obj.GroupVersionKind().Kind,
-			Name:      obj.GetName(),
-			Namespace: obj.GetNamespace(),
-		}
-		status.SelectedResources = append(status.SelectedResources, ri)
-	}
+func (r *Reconciler) setPlacementStatus(ctx context.Context, crp *fleetv1beta1.ClusterResourcePlacement, selectedResources []fleetv1beta1.ResourceIdentifier,
+	latestSchedulingPolicySnapshot *fleetv1beta1.ClusterSchedulingPolicySnapshot, latestResourceSnapshot *fleetv1beta1.ClusterResourceSnapshot) error {
+	crp.Status.SelectedResources = selectedResources
 	scheduledCondition := buildScheduledCondition(crp, latestSchedulingPolicySnapshot)
-	meta.SetStatusCondition(&status.Conditions, scheduledCondition)
+	meta.SetStatusCondition(&crp.Status.Conditions, scheduledCondition)
 
 	// When scheduledCondition is unknown, appliedCondition should be unknown too.
 	// Note: If the scheduledCondition is failed, it means the placement requirement cannot be satisfied fully. For example,
 	// pickN deployment requires 5 clusters and scheduler schedules the resources on 3 clusters. And the appliedCondition
 	// could be true when resources are applied successfully on these 3 clusters and the detailed the resourcePlacementStatuses
 	// need to be populated.
 	if scheduledCondition.Status == metav1.ConditionUnknown {
-		appliedCondition := metav1.Condition{
-			Status:             metav1.ConditionUnknown,
-			Type:               string(fleetv1beta1.ClusterResourcePlacementAppliedConditionType),
-			Reason:             ApplyPendingReason,
-			Message:            "Scheduling has not completed",
-			ObservedGeneration: crp.Generation,
+		conditions := []metav1.Condition{
+			{
+				Status:             metav1.ConditionUnknown,
+				Type:               string(fleetv1beta1.ClusterResourcePlacementSynchronizedConditionType),
+				Reason:             synchronizedPendingReason,
+				Message:            "Scheduling has not completed",
+				ObservedGeneration: crp.Generation,
+			},
+			{
+				Status:             metav1.ConditionUnknown,
+				Type:               string(fleetv1beta1.ClusterResourcePlacementAppliedConditionType),
+				Reason:             ApplyPendingReason,
+				Message:            "Scheduling has not completed",
+				ObservedGeneration: crp.Generation,
+			},
 		}
-		meta.SetStatusCondition(&status.Conditions, appliedCondition)
-		// skip populating detailed resourcePlacementStatus
-		return status, nil
+		crp.SetConditions(conditions...)
+		// skip populating detailed resourcePlacementStatus & work related conditions
+		// reset other status fields
+		crp.Status.PlacementStatuses = []fleetv1beta1.ResourcePlacementStatus{}
```

Comment:
empty status?

Reply:
yeah, I am not too sure about the empty placement status in this case. For example, say the new policy will schedule the resources to one or two more clusters. Some of the placement statuses actually are still valid.

I do understand that users need to figure out if any of those status are valid.